build/assets/scripts/cpu-load-balancing-runtime.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+COMMAND="$3"
+CONTAINER_ID="$4"
+
+if [ -n "$CONTAINER_ID" ]; then
+  CONTAINER_NAME=$(/bin/runc state "$CONTAINER_ID" 2> /dev/null | jq '.annotations."io.kubernetes.container.name"' -r)
+
+  # configs creates two containers, one as wrapper(POD) and a user container, we do not want to apply our logic to the wrapper  container
+  if [[ "$CONTAINER_NAME" != "POD" ]]; then
+    if [[ "$COMMAND" == "start" ]] || [[ "$COMMAND" == "kill" ]]; then
+      CGROUP_CPUSET=$(find /sys/fs/cgroup/cpuset/ -name "*$CONTAINER_ID*" | grep -v conmon)
+
+      if [ -n "$CGROUP_CPUSET" ]; then
+        # parse CPU's under the container cpuset.cpus cgroup
+        # explanation of awk magic:
+        # RS=, this tells awk to use , as the record separator.
+        # if, for example, the input is 0,2,4-6,8, then awk will see four records: 0 and 2 and 4-6 and 8
+        # FS=- this tells awk to use - as the field separator
+        # with FS set this way and if, for example, the input record consists of 2-4, then awk will see 2 as the first field and 4 as the second field.
+        # /-/ for loop will be applied for every record that contains -, and continue after to the next record
+        # if the record does not have -, it will just print the line, 1 is awk's shorthand for print-the-line.
+        for cpu in $(awk '/-/{for (i=$1; i<=$2; i++)printf "%s%s",i,ORS;next} 1' RS=, FS=- < "$CGROUP_CPUSET/cpuset.cpus"); do
```

Comment:
great doc, thanks. I just wonder if there is a shell tool that nicely prints the list of CPUs from a cpuset. It seems more and more a relatively frequent task for system tools.

Reply:
We have libraries for python and golang, but I did not aware of some tool that will be available our of the box on the CoreOS

Reply:
ack, same here. Let's keep the script and the awk magic, then.